export/exporter.go

```diff
@@ -0,0 +1,57 @@
+// Package export implements a library to create
+// and define different custom OpenCensus metrics
+// exporters.
+package export
+
+import (
+	"sync"
+	"time"
+
+	"go.opencensus.io/metric/metricexport"
+)
+
+// ExporterAgent defines the wrapper format of exporters
+// and data needed by all general exporters.
+type ExporterAgent struct {
+	metricexport.Exporter
+	ir             *metricexport.IntervalReader
+	initReaderOnce sync.Once
+}
+
+// Config defines the data format of the general
+// configurations of an exporter.
+type Config struct {
+	IncludeFilter     string
+	ReportingPeriodms int
+}
+
+// NewConfig returns a new exporter Config.
+func NewConfig(filter string, reportingPeriod int) Config {
+	return Config{
+		IncludeFilter:     filter,
+		ReportingPeriodms: reportingPeriod,
+	}
+}
+
+// a user should never have to use this explicitly. They would
+// simply instantiate an implemented exporter
+func newExporterAgent(exporter metricexport.Exporter) *ExporterAgent {
```

Comment:
are we still using the ExporterAgent?

Reply:
Yeah, since I didn't want to repeat code for the Kafka and HTTP exporter, I kept the ExporterAgent and that's what NewKafka and NewHTTP will instantiate and return since both need the IntervalReader (and to start the IntervalReader) to get the OpenCensus registry of metrics